docs/docs/bounty_reports/priority_inversion_truncation_flaw.md

```
@@ -0,0 +1,29 @@
+Flood the agent with 50-100 low-relevance, recent interactions (e.g., remember("User likes pizza."), remember("User bought shoes.")).
+Trigger a recall action that requires the master password context.
+Observed Behavior: Because the context window truncates by timestamp to avoid token bloat, the critical early memory is truncated. The agent suffers "Timeline Amnesia" and fails to recall the password constraint.
+Expected Behavior: The system should apply a recency_decay_multiplier to the relevance score. The early memory, having maximum relevance, should survive the truncation cut, while low-relevance recent memories are dropped.
+4. Proposed Architectural Solution
+To resolve this, the context window truncation logic must be updated from a pure chronological sort to a Relevance-Decay Composite Sort.
+
+Instead of:sorted_memories = sort_by_timestamp(memories)[:max_tokens]
+
+The logic should implement:
+
+import timedef calculate_memory_score(memory, current_time, decay_rate=0.01):    age = current_time - memory.timestamp    recency_factor = math.exp(-decay_rate * age)    return memory.relevance_score * recency_factor# Truncate by composite score, not purely by agesorted_memories = sorted(memories, key=lambda m: calculate_memory_score(m, current_time), reverse=True)truncated_context = sorted_memories[:max_tokens]
```

markdownlint-cli2 (0.22.1)

[warning] Spaces inside emphasis markers

(MD037, no-space-in-emphasis)